### PR TITLE
Worker and Options refactoring part 2

### PR DESCRIPTION
**Pull request by meiliang86**
This diff mainly moved worker.factory and worker.factory.option to their own files. 
Also moved some options around.

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java [hunk 1]
```diff
@@ -29,8 +29,15 @@
 
 public final class SingleWorkerOptions {
 
-  public static final class Builder {
+  public static Builder newBuilder() {
+    return new Builder();
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java:33]: then the constructor defined below must be private

File: src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java [hunk 2]
```diff
@@ -29,8 +29,15 @@
...
+  }
 
+  public static Builder newBuilder(SingleWorkerOptions options) {
+    return new Builder(options);
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java:37]: idem

**Review by mkolodezny — APPROVED** [review]
Just few minor comments. The overall looks good to me, great job!

File: src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java (hunk 1 above)
Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/worker/SingleWorkerOptions.java:33]: Good catch. Thanks!

**Pull request merged**